discussions-api/src/main/java/org/xwiki/contrib/discussions/server/DiscussionMessageRequestCreator.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.contrib.discussions.server;
+
+import java.util.Optional;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.contrib.discussions.domain.Message;
+import org.xwiki.stability.Unstable;
+
+/**
+ * Component responsible of creation of messages based on request information.
+ * The idea of this component is to be able to handle creation of messages from request send by the standard editor
+ * of messages provided in {@code discussion-macro}.
+ *
+ * @version $Id$
+ * @since 2.1
+ */
+@Unstable
```

Comment:
Not sure how to interpret that for a contrib extension in which you can keep the same major version for years.

Reply:
I'm using those for the contrib extension I contribute too using them with semver: I remove the unstable whenever I move to next major.

Reply:
> I remove the unstable whenever I move to next major.

That's what it means in XWiki Standard, but the major version change extremely rarely in most contrib extension and keeping an API forbidden (which is pretty much what @Unstable means) for years does not make much sense.

Reply:
I see your point, that's probably something to discuss on the forum since it's part of our best practices for XS and we're supposed to reuse them for contrib. Now I won't change that here since I already used that annotation in other places of the extension (when moving to 2.0), so I prefer to keep consistency for now at least. 
Also note that here Discussion is evolving a lot, so it kinds of make sense.

Reply:
> we're supposed to reuse them for contrib

When they make sense, but current `@Unstable` definition explicitly mention XWiki Standard cycles :)